ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/QueryFilterNode.java

```diff
@@ -577,6 +582,34 @@ private Object resolveKey(
                   FormatOptions.noEscape())))
           .orElse(null);
     }
+
+    private void getConstraint(
```

Comment:
This method name is a bit vague and sounds like a getter.  Maybe call it `setMostSelectiveConstraint` or something like that?  It would also be good to add a comment about what this does.

Reply:
Done